iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive;
+
+import java.util.Properties;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryState;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.mr.Catalogs;
+
+public class IcebergTableUtil {
+
+  private IcebergTableUtil() {
+
+  }
+
+  /**
+   * Load the iceberg table either from the {@link QueryState} or through the configured catalog.
+   * @param configuration a Hadoop configuration
+   * @param properties controlling properties
+   * @return
+   */
+  static Table getTable(Configuration configuration, Properties properties) {
+    // look for the table object stored in the query state. If it's null, it means the table was not loaded yet
+    // within the same query therefore we claim it through the Catalogs API and then store it in query state.
+    QueryState queryState = SessionState.get()
+        .getQueryState(configuration.get(HiveConf.ConfVars.HIVEQUERYID.varname));
+    Table table = null;
+    if (queryState != null) {
```

Comment:
Do we expect the query state to be null at any point during compilation? If so, that would result in constant reloading of the table. So maybe we should either remove the null checks to simplify the code, or at least log it if it's not there? What do you think?

Reply:
Right now we don't expect it to be null, but I'm not sure about the availability of SessionState object. I noticed this comment in the class:  `SessionState is not available in runtime and Hive.get().getConf() is not safe to call`.
So I put there this safeguard, to be more robust in case the SessionState is lost, but I agree some additional logging doesn't hurt.

Reply:
Maybe Optional<QueryState> ?

Reply:
In this case, wrapping the QueryState in an Optional doesn't add much value.